File: 4-Data_Structures/1068/bee1068.java
// 1068 - Balanço Parênteses I
import java.io.IOException;
import java.util.Scanner;

public class Main {
    
    public static boolean parenteses(String str) {
        int paren = 0;
        
        for(int i = 0; i < str.length() && paren != -1; i++) {
            if(str.charAt(i) == '(') paren++;
            else if(str.charAt(i) == ')') paren--;
        }
        
        return (paren != 0) ? false : true;
    }
 
    public static void main(String[] args) throws IOException {
        Scanner scan = new Scanner (System.in);
        
        while(scan.hasNext()) {
            String input = scan.nextLine();
            
            if(parenteses(input)) System.out.println("correct");
            else System.out.println("incorrect");
        }
        
        scan.close();
    }
 
}

File: 4-Data_Structures/1068/bee1068.c
// 1068 - Balanço Parênteses I
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool parenteses(char* str, int n) {

    int paren = 0;

    for(int i = 0; i < n-1 && paren != -1; i++) {
        if(str[i] == '(') paren++;
        else if(str[i] == ')') paren--;
    }

    return (paren != 0) ? false : true;
}

int main(void) {

    char input[1000];

    while(fgets(input, 1000, stdin) != NULL) {
        int n = strlen(input);

        if(parenteses(input, n)) printf("correct\n");
        else printf("incorrect\n");
    }

    return 0;
}